[0.7.0] `porter run` should spawn ephemeral pods

abelanger5 opened the pull request:
## Pull request type

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] If it's a backend change, tests for the changes have been added and `go test ./...` runs successfully from the root folder.
- [ ] If it's a frontend change, Prettier has been run
- [ ] Docs have been reviewed and added / updated if needed

## What is the current behavior?

`porter run...` connects to existing pods, which is not desired in many cases. 
 
## What is the new behavior?

`porter run...` copies the configuration for a different pod, and uses that pod to run commands in. Using the flag `--existing_pod` will provide the old functionality.  

## Technical Spec/Implementation Notes

sunguroku commented:
Works well but a few issues I've found:
- Opening an interactive shell works, but for some reason commands that exit such as `printenv` or `echo 'hello world'` fail with the following error. I've checked that the same commands execute properly when run inside the interactive shell.
<img width="930" alt="Screen Shot 2021-08-05 at 8 35 22 PM" src="https://user-images.githubusercontent.com/65516095/128438354-9f42084f-b211-426f-a3cc-04f43e48b613.png">
- Before the `Running [CMD] for release...` prompt is shown, there is no visual indication that the command is in loading state, so it wasn't clear that the command is working. 
- Assuming this is nontrivial for it to be worth, but I missed the `If you don't see a command prompt, try pressing enter` prompt and thought it was stuck at a reconnect attempt.

abelanger5 commented:
Ok, fixed a few of these issues. 

> Opening an interactive shell works, but for some reason commands that exit such as `printenv` or `echo 'hello world'` fail with the following error. I've checked that the same commands execute properly when run inside the interactive shell.

Just made the default behavior to output the logs from this pod when this happens. I also added an option `porter logs ...` while I was at it. 

> Before the `Running [CMD] for release...` prompt is shown, there is no visual indication that the command is in loading state, so it wasn't clear that the command is working.

This is because you're running with `go run main.go` first, so it needs to compile -- when the binary is already built, this is logged immediately. 

> Assuming this is nontrivial for it to be worth, but I missed the `If you don't see a command prompt, try pressing enter` prompt and thought it was stuck at a reconnect attempt.

`kubectl` also has this behavior, and it's not very clear how to get around it. I don't think it's worth tackling this, since the issue is with an [underlying package](https://pkg.go.dev/k8s.io/kubectl@v0.21.0/pkg/util/term#TTY.Safe).
